server/src/services/customers.ts

```diff
@@ -21,6 +21,19 @@ import {customerStorage} from '../storage';
 const getCustomer = async (customerId: number): Promise<CustomerResponse> =>
   customerStorage.getCustomer(customerId);
 
+/**
+ * Gets a customer by identifying them with their data.
+ * Returns a CustomerResponse object when such a customer exits.
+ * Returns null otherwise.
+ * @param data Data of the customer that we are trying to get
+ */
+const getCustomerWithData = async (
```

Comment:
Why not just have a `getCustomerWithGpayId`? I think it's weird for the caller of this method to pass in a whole `CustomerPayload` object when you just need one property. I'm guessing you don't want to expose the uniqueness of the gpayId to the api layer? But I think it's also weird because, what happens if the data doesn't include `gpayId`? Then, based on the name, you should find customer based on the remaining properties or something like that, which is not really done here.

Reply:
@karenfrilya97  Hmm yea I did think about that before. :( But I was thinking this is more of business logic that customers are unique by their gpayId instead of something that the api layer should be aware of :/ I'm not sure what's the best way around this either.

Reply:
+1 for getCustomerWithGpayId.
gpayId is required in CustomerPayload, so you'll always pass in gpayId. But I don't see future use cases by passing gpayId+other attribute (e.g. address) together

Reply:
@ifhuang @karenfrilya97  Ok, changed